cpp/src/gandiva/jni/config_builder.cc

```diff
@@ -33,9 +33,9 @@ using gandiva::ConfigurationBuilder;
  */
 JNIEXPORT jlong JNICALL
 Java_org_apache_arrow_gandiva_evaluator_ConfigurationBuilder_buildConfigInstance(
-    JNIEnv* env, jobject configuration) {
+    JNIEnv* env, jobject configuration, jboolean optimize) {
```

Comment:
is this not backward incompatible for other users..

would it be better to introduce a overloaded method and leave the old one as true?